src/App.vue

```diff
@@ -29,23 +29,46 @@ export default {
 	data() {
 		return {
 			message: `Hello ${UserUtil.getDisplayName()}!`,
-			calendarEvents: null,
-			upcomingEvents: null,
+			calendarEvents: [],
+			upcomingEvents: [],
 		};
 	},
 	async mounted() {
+		// Upcoming events
 		const today = EventUtil.getSecondsSinceEpoch();
-		this.upcomingEvents = await EventUtil.fetchCalendarEvents(
+		const upcomingEvents = await EventUtil.fetchCalendarEvents(
 			UserUtil.getUserName(),
 			"personal",
 			today
 		);
+
+		this.upcomingEvents = this.getNextFiveEvents(upcomingEvents);
+
+		// Calendar events
 		const sixMonthsBack = EventUtil.getSecondsSinceEpochMinus6Months();
```

Comment:
This may not be an issue now, but I foresee it being in the future. Getting all events from 6 months back and onwards is going to be a very taxing operation when providers actually start using the service. To strike a good balance between performance and work needed, I think we should fetch each month on demand. Though this may be better to move to a separate card, so we can get the bug fix merged.

Getting all events from one month also isn't ideal, but I think is a fine compromise at this stage. The ideal solution would probably be having an endpoint in the app, so we can optimise the fetching in the backend. But I think that should be a future optimisation.

The same I can see applies to the upcoming events, I think we're going to have to limit it somewhere in the 2 weeks to a month range, and then just display a message saying, you don't have any events in the next 2 weeks.

Reply:
Should be solved now.

Reply:
> We still need a limit on how far ahead the upcoming events are fetched, but is that a task for @oliveiraigorm ?

There is a default setting on this.

```
	getApiUrl: (userName, calendarName, fromDate, endDate) => {
		let url = `/remote.php/dav/calendars/${userName}/${calendarName}?export&expand=1&accept=jcal&componentType=VEVENT&start=${fromDate}`;
		url += `&end=${endDate || fromDate + EventUtil.oneMonthInSeconds}`; // See this line
		return url;
	},
```